geps/gep-1713/index.md

```diff
@@ -0,0 +1,408 @@
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: httproute-example
+spec:
+  parentRefs:
+  - name: second-workload-listeners
+    kind: ListenerSet
+    sectionName: second
+```
+
+For instance, the following `HTTPRoute` attemps to attach to a listener defined in the parent `Gateway`. This is not valid and the route's status should reflect that.
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: httproute-example
+spec:
+  parentRefs:
+  - name: some-workload-listeners
+    kind: ListenerSet
+    sectionName: foo
+```
+
+#### Listener Validation
+
+Implementations MUST treat the parent `Gateway` as having the concatenated list of all listeners from itself and attached `ListenerSets`
+Validation of this list of listeners MUST behave the same as if the list were part of a single `Gateway`.
+
+From the earlier example the above resources would be equivalent to a single `Gateway` where the listeners are collapsed into a single list.
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: parent-gateway
+spec:
+  gatewayClassName: example
+  listeners:
+  - name: foo
+    hostname: foo.com
+    protocol: HTTP
+    port: 80
+  - name: first
+    hostname: first.foo.com
+    protocol: HTTPS
+    port: 443
+    tls:
+      mode: Terminate
+      certificateRefs:
+      - kind: Secret
+        group: ""
+        name: first-workload-cert # Provisioned via HTTP01 challenge
+  - name: second
+    hostname: second.foo.com
+    protocol: HTTPS
+    port: 443
+    tls:
+      mode: Terminate
+      certificateRefs:
+      - kind: Secret
+        group: ""
+        name: second-workload-cert # Provisioned via HTTP01 challenge
+```
+
+#### Listener Precedence
+
+Listeners should be merged using the following precedence:
+1. "parent" Gateway
+2. ListenerSet ordered by creation time (oldest first)
+3. ListenerSet ordered alphabetically by “{namespace}/{name}”.
+
+If there are listener conflicts, this should be reported as `Conflicted=True` in the ListenerStatus as usual. See 'Conditions' section below for more details on object should report the conflict.
+
+###  Gateway Conditions
+
+`Gateway` currently supports the following top-level condition types: `Accepted` and `Programmed`
+
+For a `Gateway`, `Accepted` should be set based on the entire set of merged listeners.
+For instance, if a `ListenerSet` listener is invalid, `ListenersNotValid` would be reported. 
+`Programmed` is not expected, generally, to depend on the children resources, but if an implementation does depend on these
+they should consider child resources when reporting this status.
+
+Parent gateways MUST NOT have `ListenerSet` listeners in their `status.listeners` conditions list.
+
+It is up to the implementation whether an invalid listener affects other listeners in the Gateway.
```

Comment:
Nope, it can't work like this or it will be not portable.

If an implementation supports the `ListenerSet` feature, then it MUST have invalid Listeners affect other listeners either in or attached to the Gateway, in the same way as they would if all the Listeners were in one Gateway. (This also links again to the discussion about whether we should make Listeners optional).

Reply:
+1. Maybe the approach would need to look something like this?

https://github.com/kubernetes-sigs/gateway-api/blob/86c06a0b62587a627a9929d5943c5894a608ad62/apis/v1/shared_types.go#L401-L415

Reply:
> Maybe the approach would need to look something like this?

I removed the problematic line.  I don't think we need a new 'partial' condition because we have `ListenerStatus` that can surface details for problematic listeners.